Source/ACE/Network/GameAction/Actions/GameActionChangeCombatMode.cs

```diff
@@ -9,7 +10,7 @@ public static class GameActionChangeCombatMode
         public static void Handle(ClientMessage message, Session session)
         {
             var newCombatMode = message.Payload.ReadUInt32();
-            session.Player.SetCombatMode((CombatMode)newCombatMode);
+            session.Player.SetCombatMode((CombatMode)newCombatMode, session.Player);
```

Comment:
This is not going to be safe -- need a HandleActionSetCombatMode function for concurrency correctness -- really that whole function construction looks hacky to me -- Why is it in WorldObject and not Player.  Why is Player passed in (as opposed to making it virtual -- if it really should be defined in WorldObject, etc).